### PR TITLE
Ensure that Vert.x respects the top level CL

### PR DESCRIPTION
**Pull request by vietj**

### CODE REVIEW

File: src/main/java/io/vertx/core/impl/VertxThread.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
   private final TimeUnit maxExecTimeUnit;
   private long execStart;
   private ContextInternal context;
+  private ClassLoader topLevelTCCK;
```
Comment (vietj) [inline src/main/java/io/vertx/core/impl/VertxThread.java:31]: TCCL, not TCCK

**Comment by vietj** [comment]
ping @cescoffier @stuartwdouglas

**Review by cescoffier — APPROVED** [review]
LGTM

File: src/main/java/io/vertx/core/impl/VertxThread.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
   private final TimeUnit maxExecTimeUnit;
   private long execStart;
   private ContextInternal context;
+  private ClassLoader topLevelTCCL;
```
Comment (stuartwdouglas) [inline src/main/java/io/vertx/core/impl/VertxThread.java:31]: Can we also get a setter for this? 

For Quarkus dev mode we have an issue where the vert.x instance is long lived, so it hangs onto the first deployments CL, and subsequent restarts result in Vert.x having the wrong CL.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

File: src/main/java/io/vertx/core/impl/VertxThread.java (hunk 2 above)
Comment (vietj) [reply src/main/java/io/vertx/core/impl/VertxThread.java:31]: I think the top level CL is set to null when it is not executing a Vert.x stack and it is not cached

Comment (stuartwdouglas) [reply src/main/java/io/vertx/core/impl/VertxThread.java:31]: Ah, mis-read this this

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**